pandas/core/frame.py

```diff
@@ -1911,26 +1915,32 @@ def to_stata(self, fname, convert_dates=None, write_index=True,
 
         See Also
         --------
-        pandas.read_stata : Import Stata data files
-        pandas.io.stata.StataWriter : low-level writer for Stata data files
-        pandas.io.stata.StataWriter117 : low-level writer for version 117 files
+        pandas.read_stata : Import Stata data files.
+        pandas.io.stata.StataWriter : Writer for Stata data files.
+        pandas.io.stata.StataWriter117 : Writer for version 117 files.
 
         Examples
         --------
-        >>> data.to_stata('./data_file.dta')
+
+        >>> df = pd.DataFrame({'col1': [1, 2, 3], 'col2': [4, 5, 6]})
+        >>> df.to_stata('./data_file.dta')
 
         Or with dates
 
-        >>> data.to_stata('./date_data_file.dta', {2 : 'tw'})
+        >>> dates_1 = pd.to_datetime(["2016-05-06","2017-05-07", "2018-05-10"])
+        >>> dates_2 = pd.to_datetime(["2016-11-06","2017-11-07", "2018-11-10"])
+        >>> df_dates = pd.DataFrame({'date1': dates_1, 'date2': dates_2})
+        >>> df_dates.to_stata('./date_data_file.dta', {2 : 'tw'})
```

Comment:
Can you explain what the `{2: 'tw'}` means? I know this is part of the original docstring, but I think we should make the examples in a way that they explain what is going on. I don't know much about stata or this function, and this without explanation is not useful to me.

Also, would be great to simplify this. May be just a date column if that makes sense, without creating variables first, and may be using `date_range` to avoid so much information that does not add value. And if we can create a single DataFrame at the beginning that is useful for all functions, that would be much better.

Finally, if we can use something looking more like a real example. I think it gives a better impression and helps understand than a [1, 2, 3]. We've got some examples with animals in `pandas/core/generic.py`, I'd use one of them.

Reply:
The `tw` is a Stata specific option for displaying time in weeks i believe - i have updated it with  a single data frame using the animals examples from `pandas/core/generic.py` and excluded the separate example specifically for dates  (i agree it does not really add any value).